packages/js/email-editor/src/components/header/test/index.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { useSelect } from '@wordpress/data';
+import {
+	// @ts-expect-error -- It is not exported yet.
+	useEntitiesSavedStatesIsDirty,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { SendButton } from '../send-button';
+import { storeName } from '../../../store';
+import { recordEvent } from '../../../events';
+
+jest.mock( '@wordpress/data', () => ( {
+	useSelect: jest.fn(),
+	createRegistrySelector: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/core-data', () => ( {
+	createSelector: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/preferences', () => ( {
+	combineReducers: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/editor', () => ( {
+	useEntitiesSavedStatesIsDirty: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/blocks', () => ( {
+	serialize: jest.fn(),
+	parse: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/hooks', () => ( {
+	applyFilters: jest.fn( ( _, value ) => value ),
+} ) );
+
+jest.mock( '@wordpress/components', () => ( {
+	Button: ( props ) => <button { ...props }>{ props.children }</button>,
+} ) );
+
+jest.mock( '../../../events', () => ( {
+	recordEvent: jest.fn(),
+} ) );
+
+const useSelectMock = useSelect as jest.Mock;
+const useEntitiesSavedStatesIsDirtyMock =
+	useEntitiesSavedStatesIsDirty as jest.Mock;
+const recordEventMock = recordEvent as jest.Mock;
+
+const mockStoreValues = {
+	hasEmptyContent: false,
+	isEmailSent: false,
+};
+
+describe( 'SendButton', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
```

Comment:
I see we create a bunch of mocks at the beginning (in later commits, move to shared-mocks), but first thing we do we clearAllMocks. Could you please elaborate why it is like that? It looks confusing to me.

Reply:
I think the function does something other than you expect.

The function [`clearAllMocks`](https://jestjs.io/docs/jest-object#jestclearallmocks) clears the count of calls, return values, etc. 
Another example, where it is used, is in [the product editor tabs test](https://github.com/woocommerce/woocommerce/blob/85a07a886da39e86ce0f76fd3b27d17620129a13/packages/js/product-editor/src/components/tabs/test/tabs.spec.tsx#L104)
Yes, those tests work without this call, and I can remove it if you want.

Reply:
Aha. TIL. Now it makes perfect sense. Thank you for the explanation.